src/settings/ServicePointDetails.js

```diff
@@ -225,76 +236,64 @@ class ServicePointDetails extends React.Component {
-                className="col-sm-1"
-              >
-                <KeyValue
-                  label={<FormattedMessage id="ui-calendar.sunDayShort" />}
-                  value={this.getWeekdayOpeningHours(weekdays[0])}
-                />
-              </div>
-              <div
-                data-test-monday
-                className="col-sm-1"
-              >
-                <KeyValue
-                  label={<FormattedMessage id="ui-calendar.monDayShort" />}
-                  value={this.getWeekdayOpeningHours(weekdays[1])}
-                />
-              </div>
-              <div
-                data-test-tuesday
-                className="col-sm-1"
-              >
-                <KeyValue
-                  label={<FormattedMessage id="ui-calendar.tueDayShort" />}
-                  value={this.getWeekdayOpeningHours(weekdays[2])}
-                />
-              </div>
-              <div
-                data-test-wednesday
-                className="col-sm-1"
-              >
-                <KeyValue
-                  label={<FormattedMessage id="ui-calendar.wedDayShort" />}
-                  value={this.getWeekdayOpeningHours(weekdays[3])}
-                />
-              </div>
-              <div
-                data-test-thursday
-                className="col-sm-1"
-              >
-                <KeyValue
-                  label={<FormattedMessage id="ui-calendar.thuDayShort" />}
-                  value={this.getWeekdayOpeningHours(weekdays[4])}
-                />
-              </div>
-              <div
-                data-test-friday
-                className="col-sm-1"
-              >
-                <KeyValue
-                  label={<FormattedMessage id="ui-calendar.friDayShort" />}
-                  value={this.getWeekdayOpeningHours(weekdays[5])}
-                />
-              </div>
-              <div
-                data-test-saturday
-                className="col-sm-1"
-              >
-                <KeyValue
-                  label={<FormattedMessage id="ui-calendar.satDayShort" />}
-                  value={this.getWeekdayOpeningHours(weekdays[6])}
-                />
-              </div>
+              {weekDays.map((weekDay) => {
+                const weekDayData = weekDaysData[weekDay];
+                const columnTestProp = `data-test-${weekDayData.columnTestProp}`;
+                const columnDynamicProps = {
+                  [columnTestProp]: true,
+                };
+
+                return (
+                  <div
+                    key={weekDay}
+                    className="col-sm-1"
+                    {...columnDynamicProps}
+                  >
+                    <KeyValue
+                      label={<FormattedMessage id={`ui-calendar.${weekDayData.labelId}`} />}
+                      value={this.getWeekdayOpeningHours(weekDay)}
+                    />
+                  </div>
+                );
+              })}
```

Comment:
This is much cleaner than the earlier implementation, and less repetitious. Thank you!

Reply:
actually this was necessary to fix issue that here first day must be based on user locale(it can be Sunday or Monday or Saturday)